test/e2e/scrapeconfig_test.go

```diff
@@ -1172,3 +1175,71 @@ var FileSDTestCases = []scrapeCRDTestCase{
 		expectedError: true,
 	},
 }
+
+var IonosSDTestCases = []scrapeCRDTestCase{
+	{
+		name: "Valid DataCeneterID",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			IonosSDConfigs: []monitoringv1alpha1.IonosSDConfig{
+				{
+					DataCenterID: "11111111-1111-1111-1111-111111111111",
+				},
+			},
+		},
+		expectedError: false,
+	},
+	{
+		name: "Invalid empty DataCenterID",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			IonosSDConfigs: []monitoringv1alpha1.IonosSDConfig{
+				{
+					DataCenterID: "",
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Invalid missing DataCenterID",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			IonosSDConfigs: []monitoringv1alpha1.IonosSDConfig{
+				{},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Valid Port number",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			IonosSDConfigs: []monitoringv1alpha1.IonosSDConfig{
+				{
+					DataCenterID: "11111111-1111-1111-1111-111111111111",
+					Port:         ptr.To(int32(8080)),
+				},
+			},
+		},
+		expectedError: false,
+	},
+	{
+		name: "Invalid Port number exceeeding the maximum value",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			IonosSDConfigs: []monitoringv1alpha1.IonosSDConfig{
+				{
+					Port: ptr.To(int32(65536)), // maximum Port number = 65535
```

Comment:
the datacenter ID needs to be set otherwise it will also the validation.
```suggestion
					DataCenterID: "11111111-1111-1111-1111-111111111111",
					Port: ptr.To(int32(65536)), // maximum Port number = 65535
```